olewoo_cs/Typelib/OWEnum.cs

```diff
@@ -40,22 +44,59 @@ public override List<ITlibNode> GenChildren()
         }
         public override void BuildIDLInto(IDLFormatter ih)
         {
+            EnterElement();
             var tde = "typedef ";
             // If the enum has a uuid, or a version associate with it, we provide that information on the same line.
-
-            if (!_ta.guid.Equals(Guid.Empty))
+            if (_data.Attributes.Count > 0)
             {
-                tde += "[uuid(" + _ta.guid + "), version(" + _ta.wMajorVerNum + "." + _ta.wMinorVerNum + ")]";
+                var lprops = _data.Attributes;
+                if (_ta.guid == Guid.Empty)
+                {
+                    var rs = lprops.Where(p => p.StartsWith("uuid")).ToList();
+                    foreach (var s in rs)
+                    {
+                        lprops.Remove(s);
+                    }
+                }
+                tde += "[" + string.Join(",", lprops) + "]";
                 ih.AppendLine(tde);
-                tde = "";
+                tde = string.Empty;
             }
-            ih.AppendLine(tde + "enum {");
+
+            ih.AppendLine(tde + "enum " + _data.ShortName + "{");
```

Comment:
your change is definitely an improvement but I don't think it's useful to have
typedef enum FOO {} FOO - 
oleview (which in the absence of better reasoning I use as canon) drops the name in favour of the typedef.

will post patch